mubench.pipeline/utils/size.py

```diff
@@ -0,0 +1,44 @@
+from sys import getsizeof, stderr
+from itertools import chain
+from collections import deque
+
+
+__handlers = {tuple: iter,
+              list: iter,
+              deque: iter,
+              dict: lambda d: chain.from_iterable(d.items()),
+              set: iter,
+              frozenset: iter,
+              }
+
+
+def total_size(o, verbose=False):
+    """ Returns the approximate memory footprint an object and all of its contents.
+
+    Automatically finds the contents of the following builtin containers and
+    their subclasses:  tuple, list, deque, dict, set and frozenset.
+    To search other containers, add handlers to iterate over their contents:
+
+        handlers = {SomeContainerClass: iter,
+                    OtherContainerClass: OtherContainerClass.get_elements}
+
+    """
+    seen = set()  # track which object id's have already been seen
+    default_size = getsizeof(0)  # estimate sizeof object without __sizeof__
```

Comment:
Do I get this right that this calculates the size of *an* object (0, in this case) as an estimate of the minimal memory footprint of any object? I was confused that this does not use `o` at first.